### DISCUSSION TITLE
pgvector support

### DISCUSSION THREAD

**Discussion by elrhomariyounes** [PostgREST/postgrest#2885] [Q&A] [answered]
Hello,

I was wondering if this api supports the [pgvector](https://github.com/pgvector/pgvector) extension.

Thank you.

**Comment by steve-chavez** [comment, marked as answer]
Sure, every pg extension is supported since you have full access to [SQL functions through RPC](https://postgrest.org/en/stable/references/api/stored_procedures.html).

Now if you're looking to use pgvector operators (`<#>`, `<=>`) directly through the API then that's not possible right now. But you can always use RPC as mentioned above.

(We're looking to extend operators on https://github.com/PostgREST/postgrest/issues/2028)

**Reply by elrhomariyounes** [reply]
Awesome thank you for the response. In a simple case where I have a table :
```sql
CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));
```
and I want to insert a record is this correct:
```bash
curl "http://localhost:3000/items" \
  -X POST -H "Content-Type: application/json" \
  -d '{ "embedding": [1,2,3] }'
```
For the operators the workaround using rpc for the moment is fine

**Reply by steve-chavez** [reply]
Yes that seems fine, inserting should work normally.

**Reply by elrhomariyounes** [reply]
Awesome thank you.

**Comment by yairyairyair** [comment]
I know it's possible via RPC but it would really help if we can also support custom operators...
It's another part to behind 100% fully possible to run any query in postgREST without some custom functions...

**Reply by yairyairyair** [reply]
Thanks in advance!
